### PR TITLE
build: Silence misc warnings

### PR DESCRIPTION
**Pull request by andreittr**
### Description of changes

This patch set silences several build warnings: inappropriate or compiler-specific flags & unused variables.

### Prerequisite checklist

 - [x] Read the [contribution guidelines](https://unikraft.org/docs/contributing/) regarding submitting new changes to the project;
 - [x] Tested your changes against relevant architectures and platforms;
 - [x] Ran the [`checkpatch.uk`](https://github.com/unikraft/unikraft/blob/staging/support/scripts/checkpatch.uk) on your commit series before opening this PR;
 - [x] Updated relevant documentation.

### Base target

 - Architecture(s): N/A
 - Platform(s): N/A
 - Application(s): N/A

### Additional configuration

N/A

### CODE REVIEW

File: Makefile.uk [hunk 1]
```diff
@@ -91,7 +91,8 @@ M4FLAGS      += -DUK_VERSION=$(UK_VERSION).$(UK_SUBVERSION)
 COMPFLAGS-$(call gcc_version_ge,6,1)	+= -fno-PIC
 LDFLAGS-$(call gcc_version_ge,6,1)	+= -no-pie
 ifeq ($(call gcc_version_ge,10,0),y)
-COMPFLAGS-y += -fhosted -fno-tree-loop-distribute-patterns
+COMPFLAGS-y += -fno-tree-loop-distribute-patterns
+CFLAGS-y += -fhosted
```
Comment (jschlumpp) [inline Makefile.uk]: As `-fhosted` is the default, I think we can also remove it (`nolibc` will set `-ffreestanding`)

Comment (andreittr) [reply Makefile.uk]: Fair point, although I would prefer to have even default options explicitly stated in our makefiles, both for documentation purposes and for the super unlikely case defaults change.

Comment (jschlumpp) [reply Makefile.uk]: My only concern is that having conflicting options for the compiler also creates a subtle dependency on the flag ordering.

**Review by rares-miculescu — APPROVED** [review]
Hello! All worked well on my side. 
Thank you, @andreittr!

Reviewed-by: Rares Miculescu <miculescur@gmail.com>

File: Makefile.uk (hunk 1 above)
Comment (andreittr) [reply Makefile.uk]: Fair point re: arg ordering; I'll remove -fhosted and leave it to be implicit.

Comment (andreittr) [reply Makefile.uk]: Updated; removed -fhosted entirely.

**Review by jschlumpp — COMMENTED** [review]
Looks good, thanks!

Reviewed-by: Marco Schlumpp <marco@unikraft.io>

**Review by razvand — APPROVED** [review]
Approved-by: Razvan Deaconescu <razvand@unikraft.io>

**Review by jschlumpp — APPROVED** [review]
Forgot to check the "Approve" radio button.

Reviewed-by: Marco Schlumpp <marco@unikraft.io>

**Pull request closed without merging**